### DISCUSSION TITLE
coreError500 on a Windows localhost

### DISCUSSION THREAD

**Discussion by bpgs** [datenstrom/community#549] [Report a bug] [answered]
I use a localhost on Windows for some test purposes.  I tried to update there from Yellow 0.8.13 to 0.8.15. The result was that at the end of the update I got a screen containing:

[editEdit] [editCreate][editDelete] myname
Datenstrom Yellow      Startseite 
[coreError500Title]
[coreError500Text]

I tried it again with installing a clean yellow 0.8.15 but the installation routine doesn't start and I get only

Datenstrom Yellow
[coreError500Title]
[coreError500Text]

on the start page. I assume that some changes from 0.8.13 to 0.8.14 (same error after starting a clean installation) are the reason.

**Comment by schulle4u** [comment]
Yes, this is a known issue. Unfortunately we couldn't track down the problem as Windows is not our prefered development environment, and thus has low priority. Please have a look at datenstrom/community#533 for possible development alternatives.

**Comment by markseuffert** [comment]
I can investigate, as soon as I have access to a Windows computer.

**Comment by BsNoSi** [comment]
Which kind of "localhost"?  

If you are using XAMPP or something there may be no solution.

A way out offers WSL2. I switched to this to overcome this. I will document the steps to get Yellow up and running on WSL2 in a few days (in German but – depending on spare time – in English, too). Before doing so, I have to test the discription for validity on another system.

**Comment by bpgs** [comment]
> 
> 
> Which kind of "localhost"?
> 
> If you are using XAMPP or something there may be no solution.
> 
> A way out offers WSL2. I switched to this to overcome this. I will document the steps to get Yellow up and running on WSL2 in a few days (in German but – depending on spare time – in English, too). Before doing so, I have to test the discription for validity on another system.

...on W7, therefore your solution will not help me.

**Comment by BsNoSi** [comment]
> ...on W7, therefore your solution will not help me.

This is correct. May I ask, why you stay on W7 which is no longer (really) maintained since years instead of moving to W10?

I was using XAMPP for years but in recent I have noticed increasing "hooks and eyelets" developing with it as `localhost`. Tests with comparable solutions to XAMPP where – in my tests – even worse. Meanwhile I accepted that Windows is not a web server and there is no "add on" to change it into a reliable web development environment. WSL is a way out because it overs a full unix subsystem that can be accessed quite easy from your Windows system. Without additional hardware or other requirements besides a more or less current running Windows 10 system.

One possible exit may be a subdomain on a/your "real" web server for development. Handling is not that handy as having it locally but you will not run into trouble because of server restrictions (except those your server sets anyway).

**Comment by BsNoSi** [comment]
***Maybe*** another reason:

When *replacing* files on the WSL server or using the update function, there is - in my case - *partially* the bizarre effect that layout files with identical contents behave differently (took me HOURS to dig it): If files have been copied or replaced, it more or less went wrong.

I then installed a fresh setup and moved content and theme files into it but  *edited* the differences in `system/layouts` and `system/settings` folder. Then it works - although the files in the end [compared by WinMerge ](https://winmerge.org/) are identical in content (and access rights, of course).

Obviously, there is an interplay between browser cache, server cache ( ⸮ *water level of the Spree, moon phase* ⸮), ... which influences this. Despite intensive search I have not found an answer - if there is any…

It is worth mentioning that it is very likely that these side effects are caused outside Yellow. Unfortunately, this does not make any difference to the personal stress level for those dealing with it…

**Comment by markseuffert** [comment]
I could reproduce the problem on Windows 10 and XAMPP 3.2.4. It was trivial to fix, please try again.

**Comment by bpgs** [comment]
> 
> 
> I could reproduce the problem on Windows 10 and XAMPP 3.2.4. It was trivial to fix, please try again.

Yeah, you killed the bug, it works (I haven't seen this **\** ).  A clean installations works without any issues. An update from 0.8.13 to 0.8.16 is a little bit tricky because you need to do it in two steps. Perhaps the reason is that some extensions are not updated with the first step. But that is not problem for me. Thanks a lot. I'll praise Yellow whereever I can.

**Comment by markseuffert** [comment, marked as answer]
You're right, I also applied the bugfix to the first step. Update from 0.8.13 to 0.8.16 should be working now.

**Reply by nzyeehaa** [reply]
0.8.14 / 0.8.15 / 0.8.16 wasnt working as of yesterday on clean install. 
0.8.17 works perfectly
I am intrigued as to what the fix was?

**Reply by markseuffert** [reply]
A character in the string `".*\.txt"` was wrongly interpreted as path separator on Windows and then ruined language files loading. The bug was introduced during code refactoring and didn't affect Linux and Mac. Fortunately my son got a gaming computer to play Fortnite, I got access to a Windows 10 machine and could fix it.

**Reply by nzyeehaa** [reply]
thanks mark, i was on the right track, just dont know my way around your code well enough. 
good work
